internal/contentcache/contentcache.go

```diff
@@ -127,27 +165,31 @@ func (c *ContentCache) NewTempFile(rc io.ReadCloser) (gcsx.TempFile, error) {
 }
 
 // AddOrReplace creates a new cache file or updates an existing cache file
-func (c *ContentCache) AddOrReplace(cacheObjectKey *CacheObjectKey, generation int64, rc io.ReadCloser) (gcsx.TempFile, error) {
+func (c *ContentCache) AddOrReplace(cacheObjectKey *CacheObjectKey, generation int64, rc io.ReadCloser) (*CacheObject, error) {
 	if cacheObject, exists := c.fileMap[*cacheObjectKey]; exists {
 		cacheObject.Destroy()
 	}
-	metadata := &gcsx.TempFileObjectMetadata{
-		BucketName: cacheObjectKey.BucketName,
-		ObjectName: cacheObjectKey.ObjectName,
-		Generation: generation,
+	file, err := c.NewCacheFile(rc)
+	metadata := &CacheFileObjectMetadata{
+		CacheFileNameOnDisk: file.Name(),
+		BucketName:          cacheObjectKey.BucketName,
+		ObjectName:          cacheObjectKey.ObjectName,
+		Generation:          generation,
 	}
-	file, err := c.NewCacheFile(rc, metadata)
+	var metadataFileName string
+	metadataFileName, err = c.WriteMetadataCheckpointFile(file.Name(), metadata)
+	cacheObject := &CacheObject{MetadataFileName: metadataFileName, CacheFileObjectMetadata: metadata, CacheFile: file}
```

Comment:
1. Please use multiple lines to specify the fields
2. This line should be after the error handling